packages/cli/generation/remote-generation/remote-workspace-runner/src/runRemoteGenerationForGenerator.ts

```diff
@@ -161,3 +177,24 @@ async function computeSemanticVersion({
     }
     return response.body.version;
 }
+
+async function uploadSources({
+    context,
+    sourceUploader,
+    sourceUploads
+}: {
+    context: InteractiveTaskContext;
+    sourceUploader: SourceUploader;
+    sourceUploads: Record<FdrAPI.api.v1.register.SourceId, FdrAPI.api.v1.register.SourceUpload>;
+}): Promise<SourceConfig> {
+    if (sourceUploads == null && sourceUploader.getSourceTypes().has("protobuf")) {
+        // We only fail hard if we need to upload Protobuf source files. Unlike OpenAPI, these
+        // files are required for successful code generation.
+        context.failAndThrow("Did not successfully upload Protobuf source files.");
+    }
+
+    context.logger.debug(`Uploading source files ...`);
+    await sourceUploader.uploadSources(sourceUploads);
```

Comment:
thoughts on having this function just return the converted Fdr Source uploads?

Reply:
Sweet works for me 👍